packages/react-components/react-tags/src/components/TagContent/useTagContent.ts

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
+import type { TagContentProps, TagContentState } from './TagContent.types';
+import { useTagContext_unstable } from '../Tag/index';
+
+const tagContentAvatarSizeMap = {
+  medium: 28,
+  small: 24,
+  'extra-small': 20,
+} as const;
+
+const tagContentAvatarShapeMap = {
+  rounded: 'square',
+  circular: 'circular',
+} as const;
+
+/**
+ * Create the state required to render TagContent.
+ *
+ * The returned state can be modified with hooks such as useTagContentStyles_unstable,
+ * before being passed to renderTagContent_unstable.
+ *
+ * @param props - props from this instance of TagContent
+ * @param ref - reference to root HTMLElement of TagContent
+ */
+export const useTagContent_unstable = (props: TagContentProps, ref: React.Ref<HTMLElement>): TagContentState => {
+  const dismissible = useTagContext_unstable(ctx => ctx.dismissible);
+  const shape = useTagContext_unstable(ctx => ctx.shape);
+  const size = useTagContext_unstable(ctx => ctx.size);
+  const interactive = useTagContext_unstable(ctx => ctx.interactive);
+
+  return {
+    components: {
+      root: interactive ? 'button' : 'div',
```

Comment:
This is problematic because you still official support `div` as a valid element for this slot. 

```tsx
<Tag interactive>
  <TagContent as="div" />
</Tag>
```

The code above will break since the rendered element is a div but has no button semantics, this is what the `useAriaButtonProps` hooks is for. It might mean that you will have to throw away the result of that hook when `interactive` is false but it avoids people from doing what I just did.

Maybe @bsunderhus  has a better suggestion?

Reply:
I updated the PR, now content slot is:
```
contentButton = useAriaButtonProps(...
...
content: interactive ? (contentButton as TagState['content']) : resolveShorthand(props.content, { required: true }) 
// needed type cast :(
```
So `<Tag interactive content={{as: 'div'}}` would work. WDYT?